File: src/main/scala/com/wikibooks/spark/ch7/scala/SimpleListener.scala
package com.wikibooks.spark.ch7.scala

import org.apache.spark.sql.streaming.StreamingQueryListener
import org.apache.spark.sql.streaming.StreamingQueryListener.QueryProgressEvent
import org.apache.spark.sql.streaming.StreamingQueryListener.QueryStartedEvent
import org.apache.spark.sql.streaming.StreamingQueryListener.QueryTerminatedEvent

class SimpleListener extends StreamingQueryListener {

  override def onQueryStarted(queryStarted: QueryStartedEvent): Unit = {
    println("onQueryStarted: " + queryStarted.runId)
  }

  override def onQueryProgress(queryProgress: QueryProgressEvent): Unit = {
    println("onQueryProgress:" + queryProgress.progress.prettyJson)
  }

  override def onQueryTerminated(queryTerminated: QueryTerminatedEvent): Unit = {
    println("onQueryTerminated:" + queryTerminated.runId)
  }
}

File: src/main/java/com/wikibooks/spark/ch7/SimpleListener.java
package com.wikibooks.spark.ch7;

import org.apache.spark.sql.streaming.StreamingQueryListener;

public class SimpleListener extends StreamingQueryListener {

	@Override
	public void onQueryStarted(QueryStartedEvent queryStarted) {
		System.out.println(queryStarted.runId());
	}

	@Override
	public void onQueryProgress(QueryProgressEvent queryProgress) {
		System.out.println(queryProgress.progress());
	}

	@Override
	public void onQueryTerminated(QueryTerminatedEvent queryTerminated) {
		System.out.println(queryTerminated.runId());
	}
}